lib/util/cache.js

```diff
@@ -0,0 +1,178 @@
+'use strict'
+
+const UNSAFE_METHODS = [
+  'POST', 'PUT', 'PATCH', 'DELETE'
+]
+
+/**
+ * @see https://www.rfc-editor.org/rfc/rfc9111.html#name-cache-control
+ * @see https://www.iana.org/assignments/http-cache-directives/http-cache-directives.xhtml
+ *
+ * @typedef {{
+ *   'max-stale'?: number;
+ *   'min-fresh'?: number;
+ *   'max-age'?: number;
+ *   's-maxage'?: number;
+ *   'stale-while-revalidate'?: number;
+ *   'stale-if-error'?: number;
+ *   public?: true;
+ *   private?: true | string[];
+ *   'no-store'?: true;
+ *   'no-cache'?: true | string[];
+ *   'must-revalidate'?: true;
+ *   'proxy-revalidate'?: true;
+ *   immutable?: true;
+ *   'no-transform'?: true;
+ *   'must-understand'?: true;
+ *   'only-if-cached'?: true;
+ * }} CacheControlDirectives
+ *
+ * @param {string} header
+ * @returns {CacheControlDirectives}
+ */
+function parseCacheControlHeader (header) {
+  /**
+   * @type {import('../util/cache.js').CacheControlDirectives}
+   */
+  const output = {}
```

Comment:
prototype pollution?

Reply:
I don't believe that's possible here, the only time a key is added to the `output` object is if it's a known caching directive. If it's not, it'll be ignored